src/utils.jl

```diff
@@ -76,3 +76,10 @@ function updatemin!(r::AbstractArray, x::AbstractArray)
     end
     return r
 end
+
+function check_assignments(assignments, nclusters)
+    nclusters >= 2 || throw(ArgumentError("silhouettes() not defined for the degenerated clustering with a single cluster."))
```

Comment:
Ah, this line is `silhouettes()`-specific, it should be moved out of generic checking function to `silhouettes()` code.